src/app/modules/submission/encrypt-submission/encrypt-submission.middleware.ts

```diff
@@ -184,17 +190,48 @@ export const validateSubmission = async (
         req.formsg.featureFlags.includes(featureFlags.encryptionBoundaryShift),
```

Comment:
In the preceeding middleware `EncryptSubmissionMiddleware.checkNewBoundaryEnabled`, we already return an error status if `encryptionBoundaryShift` feature flag is not enabled. So `req.formsg.featureFlags.includes(featureFlags.encryptionBoundaryShift)` should always be true (same as what we do in L184 of `email-submission.controller`)

Mainly coming from the view that passing in `req.formsg.featureFlags.includes(featureFlags.encryptionBoundaryShift)` is a bit untidy; if we mistakenly disable `encryptionBoundaryShift` feature flag but the response gets to this middleware, then the storage mode submission will be processed as if it were an encrypted mode submission, and there may be unexpected outcomes because the submission shape is not what we expect.

Reply:
True, thanks! I've removed the redundant check in [731a04b](https://github.com/opengovsg/FormSG/pull/6696/commits/731a04b015acdc300add49fdb2a6ad7ad8a799b7). 

I've done some refactoring to remove the `encryptionBoundaryShiftEnabled` param from `parseResponses` altogether as it is only used for processing submissions in the clear. Now `getFilteredResponses` simply takes in a param `isEncryptedMode` to get the right filters.